### PR TITLE
quick fix to filter duplicated spot of cost and base line series. but…

### PR DESCRIPTION
**Pull request by snakewa**
… still need to find out why positionAverageCostHistoryToLineData() and positionBaseHistoryToLineData() will produce thses dots

### CODE REVIEW

**Comment by snakewa** [comment]
polygon:0x11AE25B81A681479A0BeA0e32a22505214F9F4c2

**Comment by c9s** [comment]
these data are triggered by the klines, some klines are duplicated, we need to fix it from the golang side.

**Comment by c9s** [comment]
these position history data are huge, would suggest not to iterate all of them. (we have already iterate it once 😂

File: apps/backtest-report/components/TradingViewChart.js [hunk 1]
```diff
@@ -398,14 +398,18 @@ const TradingViewChart = (props) => {
       if (chartData.positionHistory) {
         const lineSeries = chart.current.addLineSeries();
         const costLine = positionAverageCostHistoryToLineData(currentInterval, chartData.positionHistory);
-        lineSeries.setData(costLine);
+        const costLineIdx = {}
+        const costLineDeduplicated = costLine.filter( v => { const exists = costLineIdx[v.time]; costLineIdx[v.time] = v.value; return !exists  });
```
Comment (c9s) [inline apps/backtest-report/components/TradingViewChart.js:402]: move this check to the positionAverageCostHistoryToLineData function? there is already a for loop.

File: apps/backtest-report/components/TradingViewChart.js [hunk 2]
```diff
@@ -398,14 +398,18 @@ const TradingViewChart = (props) => {
...
+        lineSeries.setData(costLineDeduplicated);
 
         const baseLineSeries = chart.current.addLineSeries({
           priceScaleId: 'left',
           color: '#98338C',
         });
         const baseLine = positionBaseHistoryToLineData(currentInterval, chartData.positionHistory)
-        baseLineSeries.setData(baseLine);
+        const baseLineIdx = {}
+        const baseLineDeduplicated = baseLine.filter( v => { const exists = baseLineIdx[v.time]; baseLineIdx[v.time] = v.value; return !exists  });
```
Comment (c9s) [inline apps/backtest-report/components/TradingViewChart.js:411]: move this check to the positionBaseHistoryToLineData function? there is already a for loop.

File: apps/backtest-report/components/TradingViewChart.js (hunk 1 above)
Comment (snakewa) [reply apps/backtest-report/components/TradingViewChart.js:402]: sure. let me look into the function

**Comment by zenixls2** [comment]
any update or action required for this pr? @c9s @snakewa 
if not, maybe we could close the pr.

**Pull request closed without merging**